src/js/controllers/mapController.ts

```diff
@@ -1183,6 +1188,289 @@ export class MapController {
+    endDate: string | Date
+  ): any {
+    startDate = new Date(startDate);
+    endDate = new Date(endDate);
+
+    if (this._map) {
+      const layer = (this._map.allLayers as any).items.filter(
+        (layer: any) => layer.id === layerID
+      );
+
+      const whereClause = `ACQ_DATE > date '${startDate}' AND ACQ_DATE < date '${endDate}'`;
+      /**
+       * TODO
+       * Need to update definitionExpression of the layer's selected sublayer
+       * Check layer.capabilities.exportMap.supportsSublayerDefinitionExpression to confirm we can update definitionExpression
+       */
+    }
+  }
+
+  initializeAndSetVIIRSLayers(): void {
+    const viirsLayerIDs = VIIRSLayerIDs.map(({ layerID, url }) => {
+      return new FeatureLayer({
+        id: layerID,
+        url,
+        visible: false
+      });
+    });
+
+    this._map?.addMany(viirsLayerIDs);
+  }
+
+  initializeAndSetMODISLayers(): void {
+    const modisLayerIDs = MODISLayerIDs.map(({ layerID, url }) => {
+      return new FeatureLayer({
+        id: layerID,
+        url,
+        visible: false
+      });
+    });
+
+    this._map?.addMany(modisLayerIDs);
+  }
+
+  setMODISDefinedRange(layer: any, sublayerType: string): void {
+    if (!this._map) {
+      return;
+    }
+
+    const MODIS24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+
+    switch (sublayerType) {
+      case '24 hrs':
+        {
+          MODIS24.visible = true;
+          MODISLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+
+            if (specificLayer) {
+              specificLayer.visible = false;
+            }
+          });
+        }
+        break;
+      case '48 hrs':
+        {
+          MODIS24.visible = false;
+          MODISLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+            if (specificLayer) {
+              if (specificLayer.id.includes('48')) {
+                specificLayer.visible = true;
+              } else {
+                specificLayer.visible = false;
+              }
+            }
+          });
+        }
+        break;
+      case '72 hrs':
+        {
+          MODIS24.visible = false;
+          MODISLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+            if (specificLayer) {
+              if (specificLayer.id.includes('72')) {
+                specificLayer.visible = true;
+              } else {
+                specificLayer.visible = false;
+              }
+            }
+          });
+        }
+        break;
+      case '7 days':
+        {
+          MODIS24.visible = false;
+          MODISLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+            if (specificLayer) {
+              if (specificLayer.id.includes('7d')) {
```

Comment:
can we do strict equality comparison again? I dont know if this can cause bugs, we are partially matching id name with two character string when we know that we need layer which ID is `MODIS7D` ?